fieldservice_person_salary_sale_reference/models/fsm_order_salary.py

```diff
@@ -0,0 +1,12 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import models, fields
+
+
+class FSMOrderSalary(models.Model):
+    _inherit = "fsm.order.salary"
+
+    sale_reference_salary = fields.Char(
+        related="fsm_order_id.sale_id.client_order_ref", 
```

Comment:
This addon fieldservice_sale_reference adds related reference field then we can do this, much simpler
https://github.com/solvosci/slv-field-service/blob/8bd7bd9f7cf92ae3780026c2557a0c0ed23e59d9/fieldservice_sale_reference/models/fsm_order.py#L10
```suggestion
        related="fsm_order_id.sale_reference", 
```

Reply:
Comment attended